File: src/motion-profiles.py
import numpy as np
import matplotlib.pyplot as plt

def velocity(t, A=1, f=0.5):
	return A * 1/(2 * np.pi * f) * (1 - np.cos(2 * np.pi * f * t))

def acceleration(t, A=1, f=0.5):
	return A * np.sin(2 * np.pi * f * t)	


t = np.linspace(0, 2, 100)

plt.plot(t, velocity(t))
plt.plot(t, acceleration(t))
plt.show()

File: src/motion-profiles.jl
function velocity(t::Array{Float64,1}; A::Number = 1, f::Number = 0.5)
	@. A * 1/(2 * π * f) * (1 - cos(2 * π * f * t))
end
    
function acceleration(t::Array{Float64,1}; A::Number = 1, f::Number = 0.5)
	@. A * sin(2 * π * f * t)
end
    
function displacement(A::Number; T::Number = 1)	
	A * T^2 /(2π)
end 
    
function amplitude(θ::Number; T::Number = 1)	
	(2π/T^2) * θ
end
    
function peak_velocity(A::Number; T::Number = 1)	
	(T/π) * A
end